src/components/orderCreator/OrderCreator.jsx

```diff
@@ -104,8 +118,17 @@ function OrderCreator() {
           ))}
         </div>
       ) : (
-        <MenuPreview />
+        // view 2, select menu type
+        <MenuPreview
+          counter1={counter1}
+          setCounter1={setCounter1}
```

Comment:
Al pasar de aquí orderCreator a menuPreview sale un warning --> Cannot update a component (`OrderCreator`) while rendering a different component (`MenuPreview`).

Creo que no puedes pasarel el setState directamente como prop, intenta levantar el estado metiendo setCounter dentro de una función en orderCreator y pasando esa función a menuPreview. Ojo no estoy seguro de qu ese sea el problema pero tiene pinta

https://stackoverflow.com/questions/55719293/lifting-state-with-hooks-from-a-mapped-array#:~:text=It%27s%20easy%20buddy.%20check%20below%20code.%20I%20just%20made%20some%20changes%20in%20you%20code.